src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -0,0 +1,193 @@
+  {
+    panelTitle: 'Main schema',
+    controls: ['dimensions'],
+    children: [
+      {
+        panelTitle: 'Title',
+        controls: ['title'],
+      },
+      {
+        panelTitle: 'Font',
+        controls: ['font'],
+      },
+    ],
+  },
+  {
+    panelTitle: 'Axes and margins',
+    controls: ['axes'],
+  },
+  {
+    panelTitle: 'Colours',
+    controls: ['colourScheme', 'colourInversion'],
+  },
+  {
+    panelTitle: 'Legend',
+    controls: [
+      {
+        name: 'legend',
+        props: {
+          option: {
+            positions: 'top-bottom',
+          },
+        },
+      },
+    ],
+  },
+];
+
+const dotPlot = (props) => {
+  const { experimentId } = props;
+
+  const dispatch = useDispatch();
+  const config = useSelector((state) => state.componentConfig[plotUuid]?.config);
+  const fetching = useSelector((state) => state.genes.properties.views[plotUuid]?.fetching);
+  const cellSets = useSelector((state) => state.cellSets);
+  const highestDispersionGenes = useSelector(
+    (state) => state.genes.properties.views[plotUuid]?.data,
+  );
+
+  const PROPERTIES = ['dispersions'];
+  const tableState = {
+    pagination: {
+      current: 1, pageSize: config?.nMarkerGenes ?? 3, showSizeChanger: true, total: 0,
+    },
+    geneNamesFilter: null,
+    sorter: { field: PROPERTIES[0], columnKey: PROPERTIES[0], order: 'descend' },
+  };
+
+  useEffect(() => {
+    dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+
+    if (cellSets.hierarchy.length === 0) dispatch(loadCellSets(experimentId));
+  }, []);
+
+  useEffect(() => {
+    if (config?.selectedGenes.length === 0 && !fetching && !highestDispersionGenes) {
+      dispatch(loadPaginatedGeneProperties(experimentId, PROPERTIES, plotUuid, tableState));
+    }
+  }, [highestDispersionGenes, config, fetching]);
+
+  useEffect(() => {
+    if (config?.selectedGenes.length === 0 && highestDispersionGenes?.length > 0) {
+      updatePlotWithChanges({ selectedGenes: highestDispersionGenes });
+      dispatch(loadGeneExpression(experimentId, highestDispersionGenes, plotUuid));
+    }
+  }, [highestDispersionGenes, config]);
+
+  if (!config) {
+    return <Skeleton />;
+  }
+
+  const updatePlotWithChanges = (obj) => {
+    dispatch(updatePlotConfig(plotUuid, obj));
+  };
+
+  const renderExtraPanels = () => (
+    <>
+      <Panel header='Gene selection' key='gene-selection'>
+        <MarkerGeneSelection
+          config={config}
+          onUpdate={updatePlotWithChanges}
+          onReset={() => updatePlotWithChanges({ selectedGenes: [] })}
+        />
+      </Panel>
+      <Panel header='Select data' key='15'>
+        {config && !cellSets.loading && !cellSets.error ? (
+          <SelectData
+            config={config}
+            onUpdate={updatePlotWithChanges}
+            cellSets={cellSets}
+          />
+        ) : <Skeleton.Input style={{ width: 200 }} active />}
+      </Panel>
+    </>
+  );
+
+  const renderPlot = () => {
+    if (!config) {
```

Comment:
this will never get called, right? Since before this we are doing 
```
  if (!config) {
    return <Skeleton />;
  }
```

after cleaning this up I think it woudl be a good idea to move all returns to the bottom of the page so these kind of problems are more easily noticeable.

Reply:
This will not be called if `config` is not available (still loading). If `config` is available, this will be called.

This is placed after the `return` by purpose so that the function is not instantiated if it's not needed (if `config` does not exist). Is that the right way to think about it?